src/popup/lib/utils.ts

```diff
@@ -62,3 +64,59 @@ export function formatNumber(
     } else return value.toExponential()
   }
 }
+
+export function useLocalStorage<T>(
+  key: string,
+  {
+    defaultValue,
+    maxAge = Infinity
+  }: Partial<{ defaultValue: T; maxAge: number }> = {}
+) {
+  const hasLocalStorage = typeof localStorage !== 'undefined'
+  type Stored = { value: T; ts: number }
```

Comment:
Let's move the type outside of the function block.

Reply:
It's very much internal to this function - never exposed outside. So, I'd prefer to keep this one in.